cli_tools/gce_vm_image_import/importer/importer.go

```diff
@@ -0,0 +1,266 @@
+		return path.ToWorkingDir(WorkflowDir+ImportWorkflow, currentExecutablePath), ""
+	}
+	return path.ToWorkingDir(WorkflowDir+ImportAndTranslateWorkflow, currentExecutablePath), getTranslateWorkflowPath(customTranWorkflow, osID)
+}
+
+func getTranslateWorkflowPath(customTranWorkflow, osID string) string {
+	if customTranWorkflow != "" {
+		return customTranWorkflow
+	}
+	return daisyutils.GetTranslateWorkflowPath(osID)
+}
+
+func buildDaisyVars(translateWorkflowPath, imageName, sourceFile, sourceImage, family, description,
+	region, subnet, network string, noGuestEnvironment bool) map[string]string {
+
+	varMap := map[string]string{}
+
+	varMap["image_name"] = strings.ToLower(imageName)
+	if translateWorkflowPath != "" {
+		varMap["translate_workflow"] = translateWorkflowPath
+		varMap["install_gce_packages"] = strconv.FormatBool(!noGuestEnvironment)
+	}
+	if sourceFile != "" {
+		varMap["source_disk_file"] = sourceFile
+	}
+	if sourceImage != "" {
+		varMap["source_image"] = fmt.Sprintf("global/images/%v", sourceImage)
+	}
+	varMap["family"] = family
+	varMap["description"] = description
+	if subnet != "" {
+		varMap["import_subnet"] = fmt.Sprintf("regions/%v/subnetworks/%v", region, subnet)
+		// When subnet is set, we need to grant a value to network to avoid fallback to default
+		if network == "" {
+			varMap["import_network"] = ""
+		}
+	}
+	if network != "" {
+		varMap["import_network"] = fmt.Sprintf("global/networks/%v", network)
+	}
+	return varMap
+}
+
+func runImport(ctx context.Context, varMap map[string]string, importWorkflowPath string, zone,
+	timeout, project, scratchBucketGcsPath, oauth, ce string, gcsLogsDisabled, cloudLogsDisabled,
+	stdoutLogsDisabled bool, kmsKey, kmsKeyring, kmsLocation, kmsProject string, noExternalIP bool,
+	userLabels map[string]string) error {
+
+	workflow, err := daisycommon.ParseWorkflow(importWorkflowPath, varMap,
+		project, zone, scratchBucketGcsPath, oauth, timeout, ce, gcsLogsDisabled,
+		cloudLogsDisabled, stdoutLogsDisabled)
+	if err != nil {
+		return err
+	}
+
+	workflowModifier := func(w *daisy.Workflow) {
+		rl := &daisyutils.ResourceLabeler{
+			BuildID: os.Getenv("BUILD_ID"), UserLabels: userLabels, BuildIDLabelKey: "gce-image-import-build-id",
+			InstanceLabelKeyRetriever: func(instance *daisy.Instance) string {
+				return "gce-image-import-tmp"
+			},
+			DiskLabelKeyRetriever: func(disk *daisy.Disk) string {
+				return "gce-image-import-tmp"
+			},
+			ImageLabelKeyRetriever: func(image *daisy.Image) string {
+				imageTypeLabel := "gce-image-import"
+				if strings.Contains(image.Image.Name, "untranslated") {
+					imageTypeLabel = "gce-image-import-tmp"
+				}
+				return imageTypeLabel
+			}}
+		rl.LabelResources(w)
+		daisyutils.UpdateAllInstanceNoExternalIP(w, noExternalIP)
+	}
+
+	return workflow.RunWithModifiers(ctx, nil, workflowModifier)
+}
+
+// Run runs import workflow.
+func Run(clientID, imageName string, dataDisk bool, osID, customTranWorkflow, sourceFile,
```

Comment:
Again, not using type for each flag makes this very unreadable. The only scenario I could see this being a good practice is if all flags have the same type.

Reply:
Agree.